views/signal.erb

```diff
@@ -108,7 +108,7 @@ d3.json(document.URL, function(error, json) {
 });
 
 $(function () {
-  d3.json('//'+ location.hostname +'/cromulent-dates', function(error, json) {
+  d3.json('//'+ location.hostname + (location.port ? ':'+location.port: '') + '/cromulent-dates', function(error, json) {
```

Comment:
does this even need to specify the host:port? Can it not be

    d3.json('/cromulent-dates', …